src/main/java/com/eternalcode/parcellockers/database/ParcelDatabaseService.java

```diff
@@ -0,0 +1,332 @@
+                            entryLocker,
+                            destinationLocker
+                    );
+                    Parcel parcel = new Parcel(
+                            UUID.fromString(rs.getString("uuid")),
+                            UUID.fromString(rs.getString("sender")),
+                            meta
+                    );
+                    this.cache.getParcels().add(parcel);
+                    parcels.add(parcel);
+                }
+                return parcels;
+
+            }
+            catch (SQLException e) {
+                Sentry.captureException(e);
+                throw new ParcelLockersException(e);
+            }
+        }).orTimeout(5, TimeUnit.SECONDS);
+    }
+
+    public CompletableFuture<Set<Parcel>> findByReceiver(UUID receiver) {
+        return CompletableFuture.supplyAsync(() -> {
+            try (Connection connection = this.dataSource.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(
+                         "SELECT * FROM `parcels` WHERE `receiver` = ?"
+                 )
+            ) {
+                statement.setString(1, receiver.toString());
+                ResultSet rs = statement.executeQuery();
+                Set<Parcel> parcels = new HashSet<>();
+
+                while (rs.next()) {
+                    ParcelLocker entryLocker = this.parcelLockerDatabaseService.findByUUID(UUID.fromString(rs.getString("entryLocker")))
+                            .join()
+                            .orElseThrow();
+
+                    ParcelLocker destinationLocker = this.parcelLockerDatabaseService.findByUUID(UUID.fromString(rs.getString("destinationLocker")))
+                            .join()
+                            .orElseThrow();
+                    ParcelMeta meta = new ParcelMeta(
+                            rs.getString("name"),
+                            rs.getString("description"),
+                            rs.getBoolean("priority"),
+                            UUID.fromString(rs.getString("receiver")),
+                            ParcelSize.valueOf(rs.getString("size")),
+                            entryLocker,
+                            destinationLocker
+                    );
+                    Parcel parcel = new Parcel(
+                            UUID.fromString(rs.getString("uuid")),
+                            UUID.fromString(rs.getString("sender")),
+                            meta
+                    );
+                    parcels.add(parcel);
+                    this.cache.getParcels().add(parcel);
+                }
+                return parcels;
+
+            }
+            catch (SQLException e) {
+                Sentry.captureException(e);
+                throw new ParcelLockersException(e);
+            }
+        }).orTimeout(5, TimeUnit.SECONDS);
+    }
+
+    public CompletableFuture<Set<Parcel>> findAll() {
```

Comment:
Mocno niewydajne, robisz select * i jeszcze kolejne zapytania pod to. Nie wiem czy nie lepiej inaczej zaprojektować klasę Parcel

Reply:
Może rozwiązaniem będzie fetchowanie tego z cache?

Reply:
Najpierw zobacz jakie dane są ci potrzebne (napisz gui logikę itd), a później pisz repozytorium żeby nie wybierać wszystkich danych tylko np. Stronę 1 z gui. Na razie robisz to tak na oko ale nie będzie to najlepsze rozwiązanie.

Reply:
Będzie to wymagać dynamicznego pisania statementów w zależności od sytuacji, zobaczymy jak to wyjdzie

Reply:
Będzie poprawione w przyszłych PR dot. kodu biznesowego, obecnie nie ma to praktycznego zastosowania